packages/components/src/components/button/button.component.ts

```diff
@@ -0,0 +1,255 @@
+import { CSSResult, html, PropertyValueMap } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import styles from './button.styles';
+import { Component } from '../../models';
+import { BUTTON_COLORS, BUTTON_TYPE, BUTTON_VARIANTS, DEFAULTS } from './button.constants';
+import { getIconNameWithoutStyle, getIconSize, isValidIconSize, isValidPillSize } from './button.utils';
+
+/**
+ * button component, which ...
+ *
+ * @tagname mdc-button
+ *
+ * @slot default - This is a default/unnamed slot
+ *
+ * @cssprop --custom-property-name - Description of the CSS custom property
+ */
+class Button extends Component {
+  @property({ type: Boolean }) active: boolean = false;
+
+  @property({ type: Boolean }) disabled: boolean = false;
```

Comment:
Q: If we assign a value to a type when declaration, do we need to still need to specify the type.
```suggestion
  @property({ type: Boolean }) disabled = false;
```